docs/docs/authentication-tutorial.md

```diff
@@ -2,7 +2,445 @@
 title: Making a site with user authentication
 ---
 
-This is a stub. Help our community expand it.
+Sometimes, you need to create a site with gated content, available only to authenticated users. Using Gatsby, you may achieve this using the concept of [client-only routes](https://www.gatsbyjs.org/docs/building-apps-with-gatsby/#client-only-routes), to define which pages a user can view only after logging in.
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Prerequisites
+
+You should have already configured your environment to be able to use the `gatsby-cli`. A good starting point is the [main tutorial](https://www.gatsbyjs.org/tutorial/).
+
+## Security notice
+
+In production, you should use a tested and robust solution to handle the authentication. [Auth0](https://www.auth0.com), [Firebase](https://firebase.google.com), and [Passport.js](passportjs.org) are good examples. This tutorial will only cover the authentication workflow, but you should take the security of your app as seriously as possible.
+
+## Building your Gatsby app
+
+Start by creating a new Gatsby project:
+
+```shell
+gatsby new gatsby-auth
+cd gatsby-auth
+```
+
+Then, add a more apt title to your newly created site, changing the content of `gatsby-config.js`:
+
+```javascript:title=gatsby-config.js
+module.exports = {
+  siteMetadata: {
+    title: "Gatsby Authentication Tutorial",
+  },
+  plugins: ["gatsby-plugin-react-helmet", "gatsby-plugin-offline"],
+}
+```
+
+Create a new component to hold the links, and edit the layout component to include it. For now, it will act as a placeholder:
+
+```jsx:title=src/components/navBar.js
+import React from "react"
+import { Link } from "gatsby"
+
+export default () => (
+  <div
+    style={{
+      display: "flex",
+      flex: "1",
+      justifyContent: "space-between",
+      borderBottom: "1px solid #d1c1e0",
+    }}
+  >
+    <span>You are not logged in</span>
+
+    <nav>
+      <Link to="/">Home</Link>
+      {` `}
```

Comment:
nit: but these spaces should probably just be removed. I think it's an artifact from prettier, but a hard line break should work here instead of spacing!

Reply:
Actually, they are there on purpose. The minification process removes any whitespace preceeded by a newline, so all the links stay side by side (`<a>...</a><a>...</a>`), and end up rendering as one big word ([Home]()[Profile]()[Logout]()) - the links would work normally, but visually they wolud look like one.

I could solve this with styling (inline styles or change the global css file), but thought it could be more code than needed to explain the tutorial (or maybe I'm a lazy typist :laughing: )

Opinions?